src/Controls/StreetControls.js

```diff
@@ -102,19 +239,123 @@ class StreetControls extends FirstPersonControls {
     }
 
     /**
-     * Move the camera
+     * Method called when user click on the ground.</br>
+     * Note that this funtion contains default values that can be overrided, by overriding this function.
+     *
+     * @param {THREE.Vector3} position - The position
+     */
+    onClickOnGround(position) {
+        // compute a new position to go : 4 meter up from the picked position on the ground
+        const up = position.clone().normalize();
+        position.add(up.multiplyScalar(4));
+
+        // compute time to go there
+        const distance = this.camera.position.distanceTo(position);
+        // 500 millis constant, plus an amount of time depending of the distance (but not linearly)
+        const time = 500 + Math.sqrt(distance) * 300;
+
+        // move the camera
+        this.moveCameraTo(position, time);
+    }
+
+    /**
+     * Method called when user click on oject that is not the ground.</br>
+     * Note that this function contains default values that can be overrided, by overriding this function.
+     *
+     * @param {THREE.Vector3} position - The position
+     */
+    onClickOnWall(position) {
+        const distance = this.camera.position.distanceTo(position);
+
+        // can't click on a wall that is at 1km distance.
+        if (distance < 1000) {
+            this.animateCameraLookAt(position, 1000);
+        }
+    }
+
+    /**
+     * Animate the camera to make it look at a position, in a given time
+     *
+     * @param { THREE.Vector3 }  position - Position to look at
+     * @param { number } time - Time in millisecond
+     */
+    animateCameraLookAt(position, time) {
+        // stop existing animation
+        this.stopTweensAnimations();
+        // prepare start point and end point
+        this.start = this.camera.clone();
+        this.end = this.camera.clone();
+        this.end.lookAt(position);
+        const factor = { t: 0 };
```

Comment:
No need for a `const`

Reply:
I don't understand what you mean. The object 'factor' is constant, even if it's content 'factor.t' is varying.